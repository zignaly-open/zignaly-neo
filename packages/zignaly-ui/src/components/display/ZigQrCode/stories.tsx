import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import ZigQrCode from ".";

export default {
  title: "Display/ZigQrCode",
  component: ZigQrCode,
} as ComponentMeta<typeof ZigQrCode>;

const Template: ComponentStory<typeof ZigQrCode> = (args) => <ZigQrCode {...args} />;

export const QRCodeExample = Template.bind({});
QRCodeExample.args = {
  url: "www.zignaly.com",
};
