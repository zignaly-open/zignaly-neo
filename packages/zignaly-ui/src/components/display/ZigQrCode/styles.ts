import styled from "styled-components";

export const QRCodeContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 200px;
  height: 200px;
  border-radius: 8px;
  background: #e1e9f0;
`;
