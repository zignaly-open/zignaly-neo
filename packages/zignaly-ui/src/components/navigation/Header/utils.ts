// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
