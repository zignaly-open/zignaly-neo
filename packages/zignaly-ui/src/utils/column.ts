import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
