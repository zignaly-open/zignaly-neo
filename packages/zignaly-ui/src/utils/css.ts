import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
