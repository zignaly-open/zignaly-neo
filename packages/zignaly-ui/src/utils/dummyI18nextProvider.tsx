import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
