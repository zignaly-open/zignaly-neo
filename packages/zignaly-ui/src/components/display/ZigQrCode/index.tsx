import { QRCodeSVG } from "qrcode.react";
import React from "react";
import { QRCodeContainer } from "./styles";
import ZigTypography from "../ZigTypography";
import { Box } from "@mui/material";

// TODO: rename to ZigQRCode
const ZigQrCode = ({
  url = "www.zignaly.com",
  label = "",
  id,
  extraInfo,
  className,
  size = 160,
  width = 200,
  height = 200,
}: {
  url?: string;
  label?: string;
  id?: string;
  extraInfo?: JSX.Element;
  className?: string;
  size?: number;
  width?: number;
  height?: number;
}) => {
  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
      gap="8px"
      className={className}
    >
      {label && (
        <ZigTypography variant="h4" color="neutral100" id={`${id}-label`}>
          {label}
        </ZigTypography>
      )}
      <QRCodeContainer width={width} height={height}>
        {extraInfo && (
          <ZigTypography
            color="neutral600"
            lineHeight="13px"
            fontSize="11px"
            marginTop="-8px"
            marginBottom="7px"
          >
            {extraInfo}
          </ZigTypography>
        )}
        <QRCodeSVG size={size} value={url} id={id} />
      </QRCodeContainer>
    </Box>
  );
};

export default ZigQrCode;
