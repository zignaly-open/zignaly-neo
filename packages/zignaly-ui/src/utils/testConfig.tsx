import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
