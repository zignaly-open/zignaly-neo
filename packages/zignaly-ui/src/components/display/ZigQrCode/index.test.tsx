import React from "react";
import ZigQrCode from ".";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/display/ZigQrCode", () => {
  describe("ZigQrCode rendering", () => {
    it("render with default url", async () => {
      const { container } = renderWithProvidersUi(<ZigQrCode id={"qr-code-test-id"} />);

      const qrCode = container.querySelector("#qr-code-test-id") as Element;
      expect(qrCode).toBeVisible();
      const qrCodeStyles = getComputedStyle(qrCode);
      expect(qrCode).toMatchSnapshot();
      expect(qrCodeStyles).toMatchSnapshot();
    });
    it("render with custom url", async () => {
      const { container } = renderWithProvidersUi(
        <ZigQrCode id={"qr-code-test-id"} url={"https://github.com/zignaly-open/zignaly-neo/"} />,
      );

      const qrCode = container.querySelector("#qr-code-test-id") as Element;
      expect(qrCode).toBeVisible();
      const qrCodeStyles = getComputedStyle(qrCode);
      expect(qrCode).toMatchSnapshot();
      expect(qrCodeStyles).toMatchSnapshot();
    });
    it("render with custom size", async () => {
      const { container } = renderWithProvidersUi(
        <ZigQrCode id={"qr-code-test-id"} width={500} height={500} />,
      );

      const qrCode = container.querySelector("#qr-code-test-id") as Element;
      expect(qrCode).toBeVisible();
      const qrCodeStyles = getComputedStyle(qrCode);
      expect(qrCode).toMatchSnapshot();
      expect(qrCodeStyles).toMatchSnapshot();
    });
    it("render with label", async () => {
      const { container } = renderWithProvidersUi(
        <ZigQrCode id={"qr-code-test-id"} label={"test-label"} className={"test-class"} />,
      );

      const qrCode = container.querySelector(".test-class") as Element;
      expect(qrCode).toBeVisible();
      const qrCodeStyles = getComputedStyle(qrCode);
      expect(qrCode).toMatchSnapshot();
      expect(qrCodeStyles).toMatchSnapshot();
    });
  });
});
