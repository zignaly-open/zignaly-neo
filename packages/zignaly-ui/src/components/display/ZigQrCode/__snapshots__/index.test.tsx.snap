// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigQrCode renders ZigQrCode render custom size 1`] = `
<svg
  height="160"
  id="qr-code-test-id"
  shape-rendering="crispEdges"
  viewBox="0 0 21 21"
  width="160"
>
  <path
    d="M0,0 h21v21H0z"
    fill="#FFFFFF"
  />
  <path
    d="M0 0h7v1H0zM9 0h1v1H9zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM9 1h3v1H9zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h2v1H8zM11 2h2v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h1v1H9zM12 3h1v1H12zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM10 4h1v1H10zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h2v1H8zM12 7h1v1H12zM0 8h3v1H0zM4 8h8v1H4zM13 8h2v1H13zM18 8h1v1H18zM2 9h2v1H2zM5 9h1v1H5zM8 9h1v1H8zM11 9h2v1H11zM14 9h3v1H14zM20,9 h1v1H20zM0 10h2v1H0zM3 10h2v1H3zM6 10h4v1H6zM11 10h2v1H11zM15 10h3v1H15zM19,10 h2v1H19zM5 11h1v1H5zM7 11h1v1H7zM12 11h1v1H12zM16 11h1v1H16zM20,11 h1v1H20zM0 12h1v1H0zM3 12h1v1H3zM6 12h4v1H6zM12 12h1v1H12zM15 12h3v1H15zM19,12 h2v1H19zM8 13h6v1H8zM15 13h2v1H15zM20,13 h1v1H20zM0 14h7v1H0zM8 14h1v1H8zM11 14h7v1H11zM19,14 h2v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h5v1H8zM17 15h1v1H17zM20,15 h1v1H20zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h2v1H8zM11 16h1v1H11zM19 16h1v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h1v1H9zM11 17h2v1H11zM14 17h3v1H14zM18 17h2v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h3v1H8zM12 18h1v1H12zM14 18h3v1H14zM20,18 h1v1H20zM0 19h1v1H0zM6 19h1v1H6zM8 19h1v1H8zM15 19h2v1H15zM19 19h1v1H19zM0 20h7v1H0zM8 20h3v1H8zM13 20h1v1H13zM17 20h1v1H17zM19,20 h2v1H19z"
    fill="#000000"
  />
</svg>
`;

exports[`components/display/ZigQrCode renders ZigQrCode render custom size 2`] = `
CSSStyleDeclaration {
  "0": "visibility",
  "_importants": {
    "visibility": undefined,
  },
  "_length": 1,
  "_onChange": [Function],
  "_values": {
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigQrCode renders ZigQrCode render custom url 1`] = `
<svg
  height="160"
  id="qr-code-test-id"
  shape-rendering="crispEdges"
  viewBox="0 0 29 29"
  width="160"
>
  <path
    d="M0,0 h29v29H0z"
    fill="#FFFFFF"
  />
  <path
    d="M0 0h7v1H0zM11 0h4v1H11zM18 0h1v1H18zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM9 1h2v1H9zM17 1h1v1H17zM20 1h1v1H20zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM12 2h1v1H12zM14 2h1v1H14zM16 2h1v1H16zM18 2h3v1H18zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM12 3h2v1H12zM15 3h3v1H15zM20 3h1v1H20zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h2v1H8zM13 4h1v1H13zM16 4h1v1H16zM18 4h3v1H18zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM11 5h2v1H11zM14 5h1v1H14zM16 5h1v1H16zM18 5h1v1H18zM20 5h1v1H20zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM10 7h2v1H10zM15 7h2v1H15zM0 8h2v1H0zM5 8h3v1H5zM9 8h1v1H9zM11 8h1v1H11zM13 8h3v1H13zM17 8h1v1H17zM19 8h1v1H19zM24 8h2v1H24zM0 9h1v1H0zM2 9h2v1H2zM5 9h1v1H5zM7 9h3v1H7zM12 9h2v1H12zM15 9h1v1H15zM17 9h3v1H17zM21 9h1v1H21zM23 9h2v1H23zM26 9h2v1H26zM0 10h1v1H0zM2 10h2v1H2zM5 10h2v1H5zM10 10h1v1H10zM15 10h2v1H15zM24 10h1v1H24zM0 11h3v1H0zM4 11h2v1H4zM7 11h3v1H7zM12 11h1v1H12zM14 11h1v1H14zM18 11h2v1H18zM21 11h1v1H21zM25 11h1v1H25zM0 12h3v1H0zM6 12h1v1H6zM8 12h2v1H8zM11 12h2v1H11zM14 12h1v1H14zM16 12h1v1H16zM22 12h2v1H22zM28,12 h1v1H28zM0 13h2v1H0zM3 13h1v1H3zM5 13h1v1H5zM13 13h1v1H13zM15 13h3v1H15zM19 13h2v1H19zM22 13h3v1H22zM27,13 h2v1H27zM1 14h1v1H1zM6 14h7v1H6zM14 14h2v1H14zM18 14h1v1H18zM20 14h2v1H20zM24 14h3v1H24zM0 15h1v1H0zM4 15h2v1H4zM8 15h4v1H8zM15 15h2v1H15zM19 15h1v1H19zM23 15h2v1H23zM26 15h1v1H26zM28,15 h1v1H28zM0 16h1v1H0zM2 16h1v1H2zM4 16h5v1H4zM10 16h2v1H10zM13 16h2v1H13zM16 16h4v1H16zM21 16h1v1H21zM23 16h1v1H23zM25 16h2v1H25zM0 17h3v1H0zM5 17h1v1H5zM7 17h1v1H7zM9 17h1v1H9zM12 17h2v1H12zM15 17h5v1H15zM22 17h3v1H22zM26,17 h3v1H26zM0 18h2v1H0zM3 18h1v1H3zM6 18h1v1H6zM8 18h4v1H8zM15 18h1v1H15zM17 18h1v1H17zM19 18h1v1H19zM24 18h2v1H24zM28,18 h1v1H28zM0 19h1v1H0zM2 19h3v1H2zM7 19h3v1H7zM12 19h1v1H12zM15 19h2v1H15zM20 19h1v1H20zM22 19h1v1H22zM0 20h1v1H0zM2 20h2v1H2zM5 20h3v1H5zM12 20h3v1H12zM17 20h1v1H17zM19 20h6v1H19zM26,20 h3v1H26zM8 21h2v1H8zM14 21h5v1H14zM20 21h1v1H20zM24 21h2v1H24zM0 22h7v1H0zM8 22h5v1H8zM14 22h2v1H14zM17 22h1v1H17zM19 22h2v1H19zM22 22h1v1H22zM24 22h3v1H24zM0 23h1v1H0zM6 23h1v1H6zM8 23h4v1H8zM14 23h3v1H14zM19 23h2v1H19zM24 23h1v1H24zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM9 24h2v1H9zM12 24h1v1H12zM16 24h1v1H16zM19 24h7v1H19zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM10 25h1v1H10zM12 25h2v1H12zM15 25h2v1H15zM19 25h3v1H19zM25 25h2v1H25zM28,25 h1v1H28zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM9 26h1v1H9zM13 26h4v1H13zM18 26h10v1H18zM0 27h1v1H0zM6 27h1v1H6zM8 27h1v1H8zM10 27h2v1H10zM18 27h1v1H18zM22 27h2v1H22zM25 27h2v1H25zM28,27 h1v1H28zM0 28h7v1H0zM8 28h2v1H8zM11 28h3v1H11zM15 28h1v1H15zM17 28h1v1H17zM19 28h1v1H19zM22 28h3v1H22zM26 28h1v1H26z"
    fill="#000000"
  />
</svg>
`;

exports[`components/display/ZigQrCode renders ZigQrCode render custom url 2`] = `
CSSStyleDeclaration {
  "0": "visibility",
  "_importants": {
    "visibility": undefined,
  },
  "_length": 1,
  "_onChange": [Function],
  "_values": {
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigQrCode renders ZigQrCode render default url 1`] = `
<svg
  height="160"
  id="qr-code-test-id"
  shape-rendering="crispEdges"
  viewBox="0 0 21 21"
  width="160"
>
  <path
    d="M0,0 h21v21H0z"
    fill="#FFFFFF"
  />
  <path
    d="M0 0h7v1H0zM9 0h1v1H9zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM9 1h3v1H9zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h2v1H8zM11 2h2v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h1v1H9zM12 3h1v1H12zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM10 4h1v1H10zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h2v1H8zM12 7h1v1H12zM0 8h3v1H0zM4 8h8v1H4zM13 8h2v1H13zM18 8h1v1H18zM2 9h2v1H2zM5 9h1v1H5zM8 9h1v1H8zM11 9h2v1H11zM14 9h3v1H14zM20,9 h1v1H20zM0 10h2v1H0zM3 10h2v1H3zM6 10h4v1H6zM11 10h2v1H11zM15 10h3v1H15zM19,10 h2v1H19zM5 11h1v1H5zM7 11h1v1H7zM12 11h1v1H12zM16 11h1v1H16zM20,11 h1v1H20zM0 12h1v1H0zM3 12h1v1H3zM6 12h4v1H6zM12 12h1v1H12zM15 12h3v1H15zM19,12 h2v1H19zM8 13h6v1H8zM15 13h2v1H15zM20,13 h1v1H20zM0 14h7v1H0zM8 14h1v1H8zM11 14h7v1H11zM19,14 h2v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h5v1H8zM17 15h1v1H17zM20,15 h1v1H20zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h2v1H8zM11 16h1v1H11zM19 16h1v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h1v1H9zM11 17h2v1H11zM14 17h3v1H14zM18 17h2v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h3v1H8zM12 18h1v1H12zM14 18h3v1H14zM20,18 h1v1H20zM0 19h1v1H0zM6 19h1v1H6zM8 19h1v1H8zM15 19h2v1H15zM19 19h1v1H19zM0 20h7v1H0zM8 20h3v1H8zM13 20h1v1H13zM17 20h1v1H17zM19,20 h2v1H19z"
    fill="#000000"
  />
</svg>
`;

exports[`components/display/ZigQrCode renders ZigQrCode render default url 2`] = `
CSSStyleDeclaration {
  "0": "visibility",
  "_importants": {
    "visibility": undefined,
  },
  "_length": 1,
  "_onChange": [Function],
  "_values": {
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigQrCode renders ZigQrCode render with label 1`] = `
<div
  class="test-class MuiBox-root css-4zozz0"
>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-vcxpot-MuiTypography-root"
    id="qr-code-test-id-label"
  >
    test-label
  </h4>
  <div
    class="css-1i2eo1v"
    height="200"
    width="200"
  >
    <svg
      height="160"
      id="qr-code-test-id"
      shape-rendering="crispEdges"
      viewBox="0 0 21 21"
      width="160"
    >
      <path
        d="M0,0 h21v21H0z"
        fill="#FFFFFF"
      />
      <path
        d="M0 0h7v1H0zM9 0h1v1H9zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM9 1h3v1H9zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h2v1H8zM11 2h2v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h1v1H9zM12 3h1v1H12zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM10 4h1v1H10zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h2v1H8zM12 7h1v1H12zM0 8h3v1H0zM4 8h8v1H4zM13 8h2v1H13zM18 8h1v1H18zM2 9h2v1H2zM5 9h1v1H5zM8 9h1v1H8zM11 9h2v1H11zM14 9h3v1H14zM20,9 h1v1H20zM0 10h2v1H0zM3 10h2v1H3zM6 10h4v1H6zM11 10h2v1H11zM15 10h3v1H15zM19,10 h2v1H19zM5 11h1v1H5zM7 11h1v1H7zM12 11h1v1H12zM16 11h1v1H16zM20,11 h1v1H20zM0 12h1v1H0zM3 12h1v1H3zM6 12h4v1H6zM12 12h1v1H12zM15 12h3v1H15zM19,12 h2v1H19zM8 13h6v1H8zM15 13h2v1H15zM20,13 h1v1H20zM0 14h7v1H0zM8 14h1v1H8zM11 14h7v1H11zM19,14 h2v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h5v1H8zM17 15h1v1H17zM20,15 h1v1H20zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h2v1H8zM11 16h1v1H11zM19 16h1v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h1v1H9zM11 17h2v1H11zM14 17h3v1H14zM18 17h2v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h3v1H8zM12 18h1v1H12zM14 18h3v1H14zM20,18 h1v1H20zM0 19h1v1H0zM6 19h1v1H6zM8 19h1v1H8zM15 19h2v1H15zM19 19h1v1H19zM0 20h7v1H0zM8 20h3v1H8zM13 20h1v1H13zM17 20h1v1H17zM19,20 h2v1H19z"
        fill="#000000"
      />
    </svg>
  </div>
</div>
`;

exports[`components/display/ZigQrCode renders ZigQrCode render with label 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "2": "justify-content",
  "3": "align-items",
  "4": "gap",
  "5": "visibility",
  "_importants": {
    "align-items": "",
    "display": "",
    "flex-direction": "",
    "gap": "",
    "justify-content": "",
    "visibility": undefined,
  },
  "_length": 6,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "display": "flex",
    "flex-direction": "column",
    "gap": "8px",
    "justify-content": "center",
    "visibility": "visible",
  },
}
`;
